src/Sarif.Viewer.VisualStudio/Tags/SarifLocationTaggerProvider.cs

```diff
@@ -44,52 +47,33 @@ internal class SarifLocationTaggerProvider : IViewTaggerProvider, ITextViewCreat
                 throw new ArgumentNullException(nameof(textView));
             }
 
-            if (buffer == null)
+            if (textBuffer == null)
             {
-                throw new ArgumentNullException(nameof(buffer));
+                throw new ArgumentNullException(nameof(textBuffer));
             }
 
-            // Note that Visual Studio's tagger aggregation expects and correctly handles null
-            // if a tagger provider does not want to provide tags.
-            if (TryCreateSarifLocationTaggerInternal(buffer, out SarifLocationTagger tagger))
+            // If for some reason the text buffer doesn't belong to the text view, then skip this.
+            if (textView.TextBuffer != textBuffer)
             {
-                return tagger as ITagger<T>;
+                return null;
             }
 
-            return null;
-        }
-
-        /// <inheritdoc/>
-        public void TextViewCreated(ITextView textView)
-        {
-            ThreadHelper.ThrowIfNotOnUIThread();
-
-            if (TryCreateSarifLocationTaggerInternal(textView.TextBuffer, out SarifLocationTagger tagger))
-            {
-                tagger.TextViewCreated(textView);
-            }
-        }
-
-        private bool TryCreateSarifLocationTaggerInternal(ITextBuffer textBuffer, out SarifLocationTagger sarifLocationTagger)
-        {
-            ThreadHelper.ThrowIfNotOnUIThread();
-
-            sarifLocationTagger = null;
-
             // The SARIF viewer needs a text buffer to have a file name in order to be able to associate a SARIF
             // result location with the file. Visual Studio allows text buffers to be created at any time with our without a filename.
             // So, if there is no file name, then do not create a tagger for this buffer.
             if (!SdkUIUtilities.TryGetFileNameFromTextBuffer(textBuffer, out _))
             {
-                return false;
+                return null;
             }
 
-            sarifLocationTagger = textBuffer.Properties.GetOrCreateSingletonProperty(delegate
-            {
-                return new SarifLocationTagger(textBuffer, this.PersistentSpanFactory);
-            });
+            SarifLocationTagger tagger = new SarifLocationTagger(textView, textBuffer, this.PersistentSpanFactory);
+
+            // Text view creation comes before the ask to create a tagger.
```

Comment:
>ask [](start = 51, length = 3)

Ultra-nit: "ask" => "request"

Reply:
This call didn't even need to exist anymore. I'm glad you commented this or we would have been listening to caret events twice!

---
In reply to: [497831993](https://github.com/microsoft/sarif-visualstudio-extension/pull/206#discussion_r497831993) [](ancestors = 497831993)